xds/src/main/java/io/grpc/xds/orca/OrcaMetricReportingServerInterceptor.java

```diff
@@ -46,27 +47,40 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/9127")
 public final class OrcaMetricReportingServerInterceptor implements ServerInterceptor {
 
-  private static final OrcaMetricReportingServerInterceptor INSTANCE =
-      new OrcaMetricReportingServerInterceptor();
+  private static volatile OrcaMetricReportingServerInterceptor instance;
 
   @VisibleForTesting
   static final Metadata.Key<OrcaLoadReport> ORCA_ENDPOINT_LOAD_METRICS_KEY =
       Metadata.Key.of(
           "endpoint-load-metrics-bin",
           ProtoUtils.metadataMarshaller(OrcaLoadReport.getDefaultInstance()));
 
-  private volatile MetricRecorder metricRecorder;
+  private final MetricRecorder metricRecorder;
 
   @VisibleForTesting
-  OrcaMetricReportingServerInterceptor() {
+  OrcaMetricReportingServerInterceptor(MetricRecorder metricRecorder) {
+    this.metricRecorder = metricRecorder;
+    OrcaMetricReportingServerInterceptor.instance = this;
   }
 
-  public static OrcaMetricReportingServerInterceptor getInstance() {
-    return INSTANCE;
+  /**
+   * Returns the server interceptor if created, otherwise {@code null}.
+   */
+  @Nullable
+  public static OrcaMetricReportingServerInterceptor getInstanceIfCreated() {
```

Comment:
This API is useless and does not replace the previous `getInstance()`. If we are going to replace the old API, it'd be with a `create()` method that always returns the same thing.

Reply:
Done. Added a `create()` method that replaces the old API `getInstance()` (in addition to the `create(MetricRecorder)` method.